tooling/minimize/tests/ub/tree_borrows/protector/foreign_write_frozen.rs

```diff
@@ -0,0 +1,19 @@
+//@ compile-flags: --minimize-tree-borrows
+
+// Check that a foreign write of a protected mutable reference with frozen permission is UB.
+
+fn foo(x: *mut i32, _y: &i32) {
+    // (x, Reserved) -> (y, P[Reserved])
+    unsafe {
+        assert!(*x == 31); // (x, Active) -> (y, P[Frozen])
+        *x = 42; // UB! Foreign Write to Protected Frozen.
+    } 
+}
```

Comment:
This method is weird. I would have written it as follows:

```suggestion
fn foo(x: *mut i32, _frozen: &i32) {
    // (x, Reserved) -> (y, Reserved) -> (_frozen, P[Frozen])
    unsafe {
        *x = 42; // UB! Foreign Write to Protected Frozen.
    } 
}
```

In particular, the `assert!` statement is not needed. Even if it was there, the comment behind it is wrong, since the assert does not change anything.